pkg/activator/net/throttler.go

```diff
@@ -541,6 +567,42 @@ func (t *Throttler) handleUpdate(update revisionDestsUpdate) {
 	}
 }
 
+func (t *Throttler) handlePubEpsUpdate(eps *corev1.Endpoints) {
+	t.logger.Infof("Public EPS updates: %#v", eps)
+
+	revN := eps.Labels[serving.RevisionLabelKey]
+	if revN == "" {
+		// Perhaps, we're not the only ones using the same selector label.
+		t.logger.Infof("Ignoring update for PublicService %s/%s", eps.Namespace, eps.Name)
+		return
+	}
+	rev := types.NamespacedName{Name: revN, Namespace: eps.Namespace}
+	if rt, err := t.getOrCreateRevisionThrottler(rev); err != nil {
+		logger := t.logger.With(zap.Any(logkey.Key, rev))
+		if k8serrors.IsNotFound(err) {
+			logger.Debug("Revision not found. It was probably removed")
+		} else {
+			logger.Errorw("Failed to get revision throttler", zap.Error(err))
+		}
+	} else {
+		rt.handlePubEpsUpdate(eps, t.ipAddress)
+	}
+}
+
+func (rt *revisionThrottler) handlePubEpsUpdate(eps *corev1.Endpoints, selfIP string) {
+	// NB: this is guaranteed to be executed on a single thread.
+	epSet, _ := endpointsToDests(eps, rt.protocol)
```

Comment:
In theory we don't need to thread the protocol through I think? Since we only care about the prefix anyway.

Reply:
But we need to extract the proper subset? I'll try to think more about this PR. Since if I purge the state in my head new ideas will start pouring in :)